packages/gen-typescript-declarations/src/verify.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved. This
+ * code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be
+ * found at http://polymer.github.io/AUTHORS.txt The complete set of
+ * contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt Code
+ * distributed by Google as part of the polymer project is also subject to an
+ * additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as ts from 'typescript';
+
+/**
+ * Compile the given declaration file paths with TypeScript and return whether
+ * compilation succeeded or failed, and a "pretty" formatted error log string.
+ *
+ * Uses a TypeScript compiler configuration suitable for web development, and
+ * strict type checking.
+ */
+export function verifyTypings(filePaths: string[]): VerifyTypingsResult {
+  const program = ts.createProgram(filePaths, compilerOptions);
+  const emitResult = program.emit();
+  const diagnostics =
+      ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
+  if (diagnostics.length > 0) {
+    return {
+      success: false,
+      errorLog:
+          ts.formatDiagnosticsWithColorAndContext(diagnostics, diagnosticsHost)
+    };
+  }
+  return {success: true, errorLog: ''};
+}
+
+/**
+ * The result of calling verifyTypings.
+ */
+export interface VerifyTypingsResult {
+  success: boolean;
+  errorLog: string;
+}
+
+const compilerOptions = {
```

Comment:
Can we assume the options here? Maybe also support reading in the tsconfig.json if it exists, so that we use the user configuration if possible.

Reply:
Nit: move the options before the usage in the function.

Reply:
> Can we assume the options here? Maybe also support reading in the tsconfig.json if it exists, so that we use the user configuration if possible.

I think it's a pretty good assumption. In a real package, it doesn't matter what tsconfig.json the author used, only the end-users one is used. So ensuring that anything generated here would work with what we think is a good tsconfig seems pretty reasonable. Could always add some override ability later.

> Nit: move the options before the usage in the function.

Done.